src/ray/object_manager/pull_manager.h

```diff
@@ -265,6 +265,21 @@ class PullManager {
     std::set<uint64_t> active_requests;
     std::set<uint64_t> inactive_requests;
 
+    void RecordTaskMetrics() const {
+      ray::stats::STATS_tasks.Record(
+          -static_cast<int64_t>(requests.size()),
```

Comment:
If the task is spilled to another node before pulling is completed (this is possible when `SpillWaitingTasks` is called IIRC), isn't this double-subtracting (or subtracting multiple times whenever spill happens) the `PENDING_NODE_ASSIGNMENT`state? Maybe we should add up these states again when they are spilled to other node via `SpillWaitingTasks`?

Reply:
The repro is not very trivial. Maybe we need to
1. add delay to concurrent object pulling (which can happen upon memory pressure, but we can use the chaos testing tool here)
2. local node has enough CPUs and not enough object store memory for additional pulling (so pulling is inactive)
3. and a remote node has enough object store memory

node 1
    Object store memory: 100MB
    CPUs: 16
node 2
    Object store memory: 100MB

request A: 1 CPU, 90MB object store memory (goes to node 1)
request B: 1 CPU, 90MB object store memory (goes to node 1)

and make request A take a long time, then this is triggered IIRC

Reply:
It doesn't really matter, since the metrics are eventually consistent...